utils.py

```diff
@@ -18,13 +20,20 @@ def func_wrapper(*args, **kwds):
             }
     return func_wrapper
 
-def invoke(func, params):
+
+@trace_xray_subsegment()
+def invoke(func, params, block=False):
     j = json.dumps(params)
     stage = os.environ["stage"]
     print("Invoking %s (stage: %s) with params: %s" % (func, stage, j))
     client = boto3.client('lambda')
-    client.invoke(FunctionName="asgard-%s-%s" % (stage, func),
-                  InvocationType="Event",
-                  Payload=bytes(j, "utf8")
-                  )
-
+    res = client.invoke(FunctionName="asgard-%s-%s" % (stage, func),
+                        InvocationType="RequestResponse" if block else "Event",
+                        Payload=bytes(j, "utf8")
+                        )
+    if block:
```

Comment:
What is this `block` argument and why do we need it? (please document)

Reply:
as you can see in the live above, it determines whether the invocation type is blocking or not. if so, I return the result.